src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_TypeInference.cs

```diff
@@ -1564,9 +1564,6 @@ static void G()
                 references: new[] { ValueTupleRef, SystemRuntimeFacadeRef },
                 parseOptions: TestOptions.Regular8);
             comp.VerifyDiagnostics(
-                // (10,9): warning CS8602: Possible dereference of a null reference.
-                //         t.y.ToString();
-                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t.y").WithLocation(10, 9),
```

Comment:
>WRN_NullReferenceReceiver [](start = 37, length = 25)

I didn't understand why this diagnostic is gone. If I remember correctly, deconstructions don't yet update nullable state, so I'd think `y` is still "maybe null" at this point.

Reply:
Will investigate when handling deconstruction.

---
In reply to: [171089477](https://github.com/dotnet/roslyn/pull/25087#discussion_r171089477) [](ancestors = 171089477)